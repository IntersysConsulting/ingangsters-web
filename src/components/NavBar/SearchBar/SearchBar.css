.btn,
.btn-searchBar {
  height: 30px;
  color: white;
}

.searchBar {
  height: 35px;
  --borderRadious: 10px;
}

.searchBar .prepend {
  background-color: #ffffff;
  border-top-left-radius: var(--borderRadious);
  border-bottom-left-radius: var(--borderRadious);
  border-color: none;
  height: 100%;
}

.searchBar .searchBox {
  width: 500px;
  font-size: 14px;
  height: 100%;
  border-width: 0;
}

.searchBar .rightSearchIcon {
  display: none;
}

.searchBar .searchButton {
  background-color: var(--green);
  border-top-right-radius: var(--borderRadious) !important;
  border-bottom-right-radius: var(--borderRadious) !important;
  border-color: var(--green);
  font-size: 14px;
  height: 100%;
}

@media screen and (max-width: 915px) {
  .searchBar .searchBox {
    width: 400px;
  }
}

@media screen and (max-width: 900px) {
  .searchBar .searchBox {
    width: 350px;
  }
}

@media screen and (max-width: 780px) {
  .searchBar .searchBox {
    width: 300px;
  }
}

@media screen and (max-width: 750px) {
  .searchBar .searchBox {
    width: 250px;
  }
}

@media screen and (max-width: 665px) {
  .searchBar .searchBox {
    width: 160px;
  }
}

@media screen and (max-width: 650px) {
  .searchBar .searchBox {
    width: 150px;
  }
}

@media screen and (max-width: 565px) {
  .searchBar .searchBox {
    width: 130px;
  }
}

@media screen and (max-width: 550px) {
  .searchBarWrapper {
    width: 95%;
  }

  .searchBar .searchBox {
    width: 80%;
    border-top-left-radius: var(--borderRadious) !important;
    border-bottom-left-radius: var(--borderRadious) !important;
  }

  .searchBar .searchButton {
    display: none;
  }

  .searchBar .rightSearchIcon {
    display: inline;
  }

  .searchBar .prepend {
    display: none;
  }

  .searchBar .append {
    background-color: #ffffff;
    border-top-right-radius: var(--borderRadious);
    border-bottom-right-radius: var(--borderRadious);
    border-color: none;
    height: 100%;
  }
}
