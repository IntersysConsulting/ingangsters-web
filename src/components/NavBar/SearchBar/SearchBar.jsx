import React from "react";
import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import { MdSearch } from "react-icons/md";
import Button from "react-bootstrap/Button";

const SearchBar = () => (
  <Form>
    <Form.Group>
      <InputGroup style={{ height: "35px" }}>
        <InputGroup.Prepend
          style={{
            backgroundColor: "#ffffff",
            borderTopLeftRadius: "5px",
            borderBottomLeftRadius: "5px",
            borderColor: null,
            height: "100%"
          }}
        >
          <MdSearch
            size="1.8em"
            color="#353535"
            style={{
              position: "relative",
              left: "50%",
              top: "50%",
              transform: "translate(-50%, -50%)"
            }}
          />
        </InputGroup.Prepend>
        <Form.Control
          type="text"
          placeholder="Search here"
          aria-describedby="inputGroupPrepend"
          name="username"
          style={{
            width: "450px",
            fontSize: "14px",
            height: "100%",
            border: "none"
          }}
        />
        <InputGroup.Append style={{}}>
          <Button
            variant="searchbar"
            style={{
              backgroundColor: "#6CC04A",
              borderTopRightRadius: "5px",
              borderBottomRightRadius: "5px",
              borderColor: "#6CC04A",
              fontSize: "14px",
              height: "100%"
            }}
          >
            Search
          </Button>
        </InputGroup.Append>
      </InputGroup>
    </Form.Group>
  </Form>
);

export default SearchBar;

// <Form.Group as={Col} md="4" controlId="validationFormikUsername">
//               <Form.Label>Username</Form.Label>
//               <InputGroup>
//                 <InputGroup.Prepend>
//                   <InputGroup.Text id="inputGroupPrepend">@</InputGroup.Text>
//                 </InputGroup.Prepend>
//                 <Form.Control
//                   type="text"
//                   placeholder="Username"
//                   aria-describedby="inputGroupPrepend"
//                   name="username"
//                   value={values.username}
//                   onChange={handleChange}
//                   isInvalid={!!errors.username}
//                 />
//                 <Form.Control.Feedback type="invalid">
//                   {errors.username}
//                 </Form.Control.Feedback>
//               </InputGroup>
//             </Form.Group>
