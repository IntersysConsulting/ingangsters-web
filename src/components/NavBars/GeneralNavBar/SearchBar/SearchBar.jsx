import React from "react";
import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import { MdSearch } from "react-icons/md";
import Button from "react-bootstrap/Button";
import "./SearchBar.css";

const SearchBar = () => (
  <Form className="searchBarWrapper">
    <Form.Group>
      <InputGroup className="searchBar">
        <InputGroup.Prepend className="prepend">
          <MdSearch
            size="1.8em"
            color="#353535"
            style={{
              position: "relative",
              left: "50%",
              top: "50%",
              transform: "translate(-50%, -50%)"
            }}
          />
        </InputGroup.Prepend>
        <Form.Control
          className="searchBox"
          type="text"
          placeholder="Search here"
          aria-describedby="inputGroupPrepend"
          name="searchTerm"
        />
        <InputGroup.Append className="append">
          <Button className="searchButton" variant="searchbar">
            Search
          </Button>
          <MdSearch
            size="1.8em"
            color="#353535"
            className="rightSearchIcon"
            style={{
              position: "relative",
              left: "50%",
              top: "50%",
              transform: "translate(-50%, -50%)"
            }}
          />
        </InputGroup.Append>
      </InputGroup>
    </Form.Group>
  </Form>
);

export default SearchBar;

// <Form.Group as={Col} md="4" controlId="validationFormikUsername">
//               <Form.Label>Username</Form.Label>
//               <InputGroup>
//                 <InputGroup.Prepend>
//                   <InputGroup.Text id="inputGroupPrepend">@</InputGroup.Text>
//                 </InputGroup.Prepend>
//                 <Form.Control
//                   type="text"
//                   placeholder="Username"
//                   aria-describedby="inputGroupPrepend"
//                   name="username"
//                   value={values.username}
//                   onChange={handleChange}
//                   isInvalid={!!errors.username}
//                 />
//                 <Form.Control.Feedback type="invalid">
//                   {errors.username}
//                 </Form.Control.Feedback>
//               </InputGroup>
//             </Form.Group>
