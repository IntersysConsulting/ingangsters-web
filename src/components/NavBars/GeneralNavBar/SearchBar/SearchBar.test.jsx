import React from "react";
import { shallow } from "enzyme";
import SearchBar from "./SearchBar";

describe("General Navbar: Search Bar", () => {
  it("renders", () => {
    const shallowRender = shallow(<SearchBar />);
    expect(shallowRender).toMatchSnapshot();
  });

  it("has both right side variants", () => {
    const wrapper = shallow(<SearchBar />);

    const searchButtonList = wrapper.find(".append > .searchButton");
    const searchIconList = wrapper.find(".append > .rightSearchIcon");

    expect(searchButtonList.length).toBe(1);
    expect(searchIconList.length).toBe(1);
  });

  it("prefixIcon", () => {
    const wrapper = shallow(<SearchBar />);

    const searchIconList = wrapper.find(".searchBar > .prepend");
    expect(searchIconList.children().length).toBe(1);
  });
});
