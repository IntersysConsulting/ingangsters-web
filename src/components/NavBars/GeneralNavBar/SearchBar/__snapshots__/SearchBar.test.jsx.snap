// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Navbar: Search Bar renders 1`] = `
<Form
  className="searchBarWrapper"
  inline={false}
>
  <FormGroup>
    <InputGroup
      className="searchBar"
    >
      <InputGroupPrepend
        className="prepend"
      >
        <MdSearch
          color="#353535"
          size="1.8em"
          style={
            Object {
              "left": "50%",
              "position": "relative",
              "top": "50%",
              "transform": "translate(-50%, -50%)",
            }
          }
        />
      </InputGroupPrepend>
      <FormControl
        aria-describedby="inputGroupPrepend"
        className="searchBox"
        name="searchTerm"
        placeholder="Search here"
        type="text"
      />
      <InputGroupAppend
        className="append"
      >
        <Button
          active={false}
          className="searchButton"
          disabled={false}
          type="button"
          variant="searchbar"
        >
          Search
        </Button>
        <MdSearch
          className="rightSearchIcon"
          color="#353535"
          size="1.8em"
          style={
            Object {
              "left": "50%",
              "position": "relative",
              "top": "50%",
              "transform": "translate(-50%, -50%)",
            }
          }
        />
      </InputGroupAppend>
    </InputGroup>
  </FormGroup>
</Form>
`;
